pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -109,63 +105,90 @@ func (m *Manager) Start() error {
 	if !m.isFeatureEnabled() {
 		return nil
 	}
+	stop, err := m.start()
+	if err != nil {
+		return err
+	}
+	go func() {
+		for {
```

Comment:
can we add a unit test somehow for this retry loop to validate that after the dbus shutdown channel is closed the retry loop will restart manager successfully?

Reply:
Before, the vendor was not updated, I sent ut as a separate pr #101932

Reply:
Thanks @wzshiming for adding this to e2e test!

I was more asking if it's possible to add some type of **unit** (as opposed to integration/e2e test) for the retry loop functionality. I'm not sure if it would be easily possible, but it would be nice if possible just to validate this retry loop.